candle-core/src/metal_backend.rs

```diff
@@ -202,8 +362,11 @@ impl BackendStorage for MetalStorage {
             Err(crate::Error::EmptyTensor { op: "reduce" }.bt())?
         }
         let dtype = if return_index { DType::U32 } else { self.dtype };
-        let mut buffer = device.new_buffer(dst_el, dtype);
-        let command_buffer = self.device.command_queue.new_command_buffer();
+        if dtype == DType::U32 {
+            crate::bail!("Implement return index reduce op");
```

Comment:
Not sure to understand this error message, could you make it more useful for the final user to diagnose the actual issue and know what to tweak?

Reply:
Oh it's the reductions like argmax which have a different return types which are not implemented yet.